autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -203,7 +205,30 @@ def preprocess(self, X):
             self.features = list(X.columns)  # TODO: add fit and transform versions of preprocess instead of doing this
         return X
 
-    def fit(self, X_train, Y_train, **kwargs):
+    def _preprocess_fit_args(self, **kwargs):
+        time_limit = kwargs.get('time_limit', None)
+        max_time_limit_ratio = self.params_aux.get('max_time_limit_ratio', 1)
+        if time_limit is not None:
+            time_limit *= max_time_limit_ratio
+        max_time_limit = self.params_aux.get('max_time_limit', None)
+        if max_time_limit is not None:
+            if time_limit is None:
+                time_limit = max_time_limit
+            else:
+                time_limit = min(time_limit, max_time_limit)
+        min_time_limit = self.params_aux.get('min_time_limit', 0)
+        if min_time_limit is None:
+            time_limit = min_time_limit
+        elif time_limit is not None:
+            time_limit = max(time_limit, min_time_limit)
+        kwargs['time_limit'] = time_limit
+        return kwargs
+
+    def fit(self, **kwargs):
+        kwargs = self._preprocess_fit_args(**kwargs)
+        self._fit(**kwargs)
```

Comment:
can consider using:
```
if time_limit > 0:
    self._fit(**kwargs)
```

Reply:
Could you clarify? This crashes when time_limit isn't passed or when time_limit is None. Are you saying that if time limit exists and is 0 or a negative value, skip training entirely regardless of the model? This is valid, but would have to be coded in a slightly different way than presented in your comment:

```
if 'time_limit' not in kwargs or kwargs['time_limit'] is None or kwargs['time_limit'] > 0:
    self._fit(**kwargs) 
```

Reply:
Added